snapcraft/cli/_errors.py

```diff
@@ -86,18 +96,26 @@ def exception_handler(  # noqa: C901
         - a snapcraft handled error occurs, debug=False so only the
           exception message is shown
     """
-    # TODO pickle the traceback if on a manged host so the actual host can deal with it.
     exc_info = (exception_type, exception, exception_traceback)
     exit_code = 1
-    is_snapcraft_error = issubclass(exception_type, errors.SnapcraftError)
+    is_snapcraft_error = (
+        issubclass(exception_type, errors.SnapcraftError)
+        and exception_type is not ProviderExecError
+    )
     is_snapcraft_reportable_error = issubclass(
         exception_type, errors.SnapcraftReportableError
     )
     is_raven_setup = RavenClient is not None
+    is_snapcraft_host = exc_info is not None and not os.path.isfile(TRACEBACK_HOST)
     is_snapcraft_managed_host = (
-        distutils.util.strtobool(os.getenv("SNAPCRAFT_MANAGED_HOST", "n")) == 1
+        os.getenv("SNAPCRAFT_BUILD_ENVIRONMENT") == "managed-host"
+    )
+    is_connected_to_tty = (
+        # used by inner instance, variable set by outer instance
+        (distutils.util.strtobool(os.getenv("SNAPCRAFT_HAS_TTY", "n")) == 1)
+        if is_snapcraft_managed_host
```

Comment:
this looks really weird when read with this indentation :confused: 
Oh black...

nothing to do about this comment, this is we airing my thoughts.

Reply:
I agree, black indents are strange sometimes.